datafusion/common/src/nested_struct.rs

```diff
@@ -215,40 +271,81 @@ mod tests {
         };
     }
 
+    fn field(name: &str, data_type: DataType) -> Field {
+        Field::new(name, data_type, true)
+    }
+
+    fn non_null_field(name: &str, data_type: DataType) -> Field {
+        Field::new(name, data_type, false)
+    }
+
+    fn arc_field(name: &str, data_type: DataType) -> FieldRef {
+        Arc::new(field(name, data_type))
+    }
```

Comment:
I don't find these very helpful. The `non_null_field` one maybe because it tells me in less verbose way what is being created but I'm okay just reading `Arc::new(...)`